alphas/alpha_utils.py

```diff
@@ -758,3 +760,24 @@ def ite(x, y: pd.DataFrame, z: pd.DataFrame) -> pd.DataFrame:
         x.fillna(0).astype(int) * y + (~x.astype(bool)).fillna(0).astype(int) * z
     except:
         raise Exception("Data cannot be substituted, boolean may not be same dimensions as dataframes")
+
+def bscore(data: pd.Series, window: int) -> pd.Series:
+    """Bollinger score over the last 'window' days."""
+    rolling_mean = data.rolling(window=window).mean()
+    rolling_std = data.rolling(window=window).std()
+    return (data - rolling_mean) / rolling_std
+
+def returns(data: pd.Series, window: int) -> pd.Series:
+    """Computes returns over a given window."""
+    return data.pct_change(periods=window) + 1
+
+def csscale(data: pd.Series) -> pd.Series:
+    """Cross-sectionally scales a series."""
+    return data / data.abs().sum()
+
+def volatility(data: pd.Series) -> pd.Series:
+    """Computes the rolling 25-day standard deviation."""
+    return data.rolling(window=25).std()
+
+def adv(data: pd.DataFrame, y: int) -> pd.Series:
```

Comment:
same as above. could you point me to where you found/got the intuition to implement this fxn?

Reply:
oh i forgot to say that i just did a rolling average of the close averages over y window, lol, but i am gonna replace this w a proper implementation from a library